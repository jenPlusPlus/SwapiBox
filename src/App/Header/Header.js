import React from 'react';
import Button from './Button/Button';
import PropTypes from 'prop-types';

const Header = ({ activeButton, numFavorites,
  getPeople, getPlanets, getVehicles}) => {
  const peopleButtonTitle = 'People';
  const planetsButtonTitle = 'Planets';
  const vehiclesButtonTitle = 'Vehicles';
  const favoritesButtonTitle = 'Favorites';

  return (
    <div className="header">
<<<<<<< HEAD
      <Button  isActive={false}
        buttonTitle={'Favorites'}
=======
      <Button  isActive={favoritesButtonTitle === activeButton}
        buttonTitle={favoritesButtonTitle}
>>>>>>> add active state to favorites button
        numFavorites={numFavorites}/>
      <Button  isActive={peopleButtonTitle === activeButton}
        buttonTitle={peopleButtonTitle}
        apiCall={getPeople}/>
      <Button  isActive={planetsButtonTitle === activeButton}
        buttonTitle={planetsButtonTitle}
        apiCall={getPlanets}/>
      <Button  isActive={vehiclesButtonTitle === activeButton}
        buttonTitle={vehiclesButtonTitle}
        apiCall={getVehicles}/>
    </div>
  );
};

Header.propTypes = {
  activeButton: PropTypes.string,
  numFavorites: PropTypes.number,
  getPeople: PropTypes.number,
  getPlanets: PropTypes.number,
  getVehicles: PropTypes.number
};

export default Header;
