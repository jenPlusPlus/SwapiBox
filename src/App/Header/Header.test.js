import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import fetchMock from 'fetch-mock';
import renderer from 'react-test-renderer';
import { withRouter } from 'react-router';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router, Route } from 'react-router-dom';

describe('Header', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(
    <Header
      activeButton={'people'}
      numFavorites={1}
      getPeople={mkFun}
      getPlanets={mkFun}
      getVehicles={mkFun}
    />);

  it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
