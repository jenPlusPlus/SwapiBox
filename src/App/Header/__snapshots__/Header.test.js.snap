// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    activeButton="people"
    getPeople={[Function]}
    getPlanets={[Function]}
    getVehicles={[Function]}
    numFavorites={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          buttonTitle="Favorites"
          isActive={false}
          numFavorites={1}
/>,
        <Button
          apiCall={[Function]}
          buttonTitle="People"
          isActive={false}
/>,
        <Button
          apiCall={[Function]}
          buttonTitle="Planets"
          isActive={false}
/>,
        <Button
          apiCall={[Function]}
          buttonTitle="Vehicles"
          isActive={false}
/>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "buttonTitle": "Favorites",
          "isActive": false,
          "numFavorites": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "apiCall": [Function],
          "buttonTitle": "People",
          "isActive": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "apiCall": [Function],
          "buttonTitle": "Planets",
          "isActive": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "apiCall": [Function],
          "buttonTitle": "Vehicles",
          "isActive": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            buttonTitle="Favorites"
            isActive={false}
            numFavorites={1}
/>,
          <Button
            apiCall={[Function]}
            buttonTitle="People"
            isActive={false}
/>,
          <Button
            apiCall={[Function]}
            buttonTitle="Planets"
            isActive={false}
/>,
          <Button
            apiCall={[Function]}
            buttonTitle="Vehicles"
            isActive={false}
/>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "buttonTitle": "Favorites",
            "isActive": false,
            "numFavorites": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "apiCall": [Function],
            "buttonTitle": "People",
            "isActive": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "apiCall": [Function],
            "buttonTitle": "Planets",
            "isActive": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "apiCall": [Function],
            "buttonTitle": "Vehicles",
            "isActive": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
